src/Event/EventFactory.cpp

```diff
@@ -1,4 +1,220 @@
 #include "EventFactory.h"
+#include "../VehicleStatus/VehicleStatus.h"
+#include "../Time/Time.h"
+#include "../Location/Location.h"
+#include "Event.h"
+#include <type_traits>
+#include <atomic>
+#include <iostream>
+#include <cmath>
+#include <thread>
 
 namespace LockFreeDispatch {
+
+
+    std::vector<Vehicle *> EventFactory::vehiclesMeetingRequirements(
+            std::vector<Vehicle *> vehicleReqts,
+            std::vector<Vehicle *> orderedVehicles) {
+
+        std::vector<Vehicle*> meetingRequirements;
+
+        for (const auto & vehicle : orderedVehicles)
+        {
+            for (const auto & reqt : vehicleReqts)
+            {
+                // TODO - should confirm the Engine/Ladder type
+                if (vehicle->getCurNumCrew() >= reqt->getCurNumCrew()
+                    && vehicle->getCurWaterVolumeLitres() >= reqt->getCurWaterVolumeLitres())
+                {
+                    meetingRequirements.push_back(vehicle);
+                }
+            }
+        }
+        return meetingRequirements;
+    }
+
+    std::vector<Vehicle *> EventFactory::selectVehicles(
+            std::vector<Vehicle *> vehicleReqts, std::vector<Vehicle *> orderedVehicles,
+            BitArray *bitArray) {
+        while (true)
+        {
+            std::vector<Vehicle*> selectedVehicles;
+            uint8_t numVehiclesRequired = vehicleReqts.size();
```

Comment:
```suggestion
            const size_t numVehiclesRequired = vehicleReqts.size();
```
This is just a stack variable so you don't gain much from compressing it into a data type of size for which you do not have a register, anyway.

Reply:
`numVehiclesRequired` is modified later in the function so I can't make it const. It is a `size_t` now though